src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -464,9 +482,101 @@ private StateEditor doTraverse(
         
         s1.incrementWeight(weight);
 
+        /*
+         * If traverse mode is WALK or BICYCLE and the request contains the parameters from data-settings.json
+         * then the weight is updated based on additional parameters (like air quality)
+         * Weight is increased for each quality parameter defined in the request.
+         */
+        boolean walkingOrBiking = traverseMode == TraverseMode.WALK || traverseMode == TraverseMode.BICYCLE;
```

Comment:
This method is already way too big. Can you please move this code into a separate method?